### ISSUE TITLE
bug: duplicate CSS because of noscript

### ISSUE THREAD

**Issue by joejordanbrown** [addyosmani/critical#381]
When using `preload` and `noscript` to load a stylesheet when JavaScript is disabled, you end up with duplicate imports of the same CSS file.

**Example:**
```html
<link as="style" rel="preload" onload="this.rel='stylesheet'" href="style.css">
<noscript>
  <link rel="stylesheet" href="style.css">
</noscript>
```
See pull requests https://github.com/addyosmani/critical/pull/379 and https://github.com/addyosmani/oust/pull/24 for further details.

**Comment by bezoerb** [comment]
@joejordanbrown: thanks for your PR. It's nearly ready to merge. The only thing missing is a test case.

**Comment by davidhellmann** [comment]
Any news here @bezoerb have the same problem since update to 2.x

**Comment by ChrissiQ** [comment]
I'm having this issue, would appreciate the fix!

**Comment by Hibrix-net** [comment]
Any news?

**Comment by bezoerb** [comment]
This should be fixed in [v3.0.1](https://github.com/addyosmani/critical/releases/tag/v3.0.1)

**Comment by bezoerb** [comment]
Closing this for now. 
Feel free to re-open if the error still exists.
